core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -363,6 +363,23 @@ private[spark] class TaskSetManager(
     None
   }
 
+  def resourceOffer(
```

Comment:
i think i created the `resourceOfferInternal` method to avoid changing all the tests that used this method.
If we agree on approach I can update all tests instead of having these two methods.

Reply:
Shall we add a comment above this method to say it's only used for test?